src/lib/services/RuleEngine.ts

```diff
@@ -7,6 +7,8 @@ export interface RuleEngine {
 
 	getCatalog(): Promise<Catalog>;
 
+	shouldEngineRun(ruleGroups: RuleGroup[], rules: Rule[], target: RuleTarget[], engineOptions: Map<string, string>): boolean;
```

Comment:
Could you add comments to the new methods? I think it would be beneficial to start commenting methods declared on interfaces, we can comment the old ones as we have time.

Reply:
Good idea. Will add them.